guides/keras_nlp/neural_machine_translation.py

```diff
@@ -0,0 +1,516 @@
+"""
+Title: English-to-Spanish translation with KerasNLP
+Author: [Abheesht Sharma](https://github.com/abheesht17/)
+Date created: 2022/05/26
+Last modified: 2022/05/26
+Description: Use KerasNLP to train a sequence-to-sequence Transformer model on the machine translation task.
+"""
+
+"""
+## Introduction
+
+KerasNLP provides building blocks for NLP (model layers, tokenizers, metrics, etc.) and
+makes it convenient to construct NLP pipelines.
+
+In this example, we'll use KerasNLP layers to build an encoder-decoder Transformer
+model, and train it on the English-to-Spanish machine translation task.
+
+This example is based on the [English-to-Spanish NMT example]
+(https://keras.io/examples/nlp/neural_machine_translation_with_transformer/) by
+[fchollet](https://twitter.com/fchollet). The original example is more low-level
+and implements layers from scratch, whereas this example uses KerasNLP to show
+some more advanced approaches, such as subword tokenization and using metrics
+to compute the quality of generated translations.
+
+You'll learn how to:
+
+- Tokenize text using `keras_nlp.tokenizers.WordPieceTokenizer`.
+- Implement a sequence-to-sequence Transformer model using KerasNLP's
+`keras_nlp.layers.TransformerEncoder`, `keras_nlp.layers.TransformerDecoder` and
+`keras_nlp.layers.TokenAndPositionEmbedding` layers, and train it.
+- Use `keras_nlp.utils.greedy_search` function to generate translations
+of unseen input sentences using the greedy decoding strategy!
+
+Don't worry if you aren't familiar with KerasNLP. This tutorial shows exactly
+how simple and easy it is to use it! So, what are you waiting for? Dive right in!
```

Comment:
Please rephrase to sound more professional-casual, e.g. "Don't worry if you aren't familiar with KerasNLP. This tutorial will start with the basics. Let's dive in."